pkg/syncer/metadatasyncer.go

```diff
@@ -906,6 +906,23 @@ func initStorageQuotaPeriodicSync(ctx context.Context, metadataSyncer *metadataS
 				StorageQuotaPeriodicSyncInstanceName, err)
 			return err
 		}
+	} else {
+		rawPatch := client.MergeFromWithOptions(
+			sqPeriodicSync.DeepCopy(),
+			client.MergeFromWithOptimisticLock{})
+		oldInterval := sqPeriodicSync.Spec.SyncIntervalInMinutes
+		if sqPeriodicSync.Spec.SyncIntervalInMinutes != int(PeriodicSyncIntervalInMin.Minutes()) {
+			sqPeriodicSync.Spec.SyncIntervalInMinutes = int(PeriodicSyncIntervalInMin.Minutes())
+		}
+		err = cnsOperatorClient.Patch(ctx, sqPeriodicSync, rawPatch)
+		if err != nil {
+			log.Errorf("initStorageQuotaPeriodicSync: failed to patch StorageQuotaPeriodicSync instance"+
+				" with name %q Error: %+v",
+				StorageQuotaPeriodicSyncInstanceName, err)
+			sqPeriodicSync.Spec.SyncIntervalInMinutes = oldInterval
+			log.Infof("initStorageQuotaPeriodicSync: Initializing periodic sync with previous interval value, %d ",
+				oldInterval)
+		}
```

Comment:
You don't need an else here, and you patch regardless of sync internal change. You should patch only if there is a mismatch.

Reply:
updated the code and removed else , will always check n sync interval and patch if found mismatch with existing interval on  CR